cmd/pinniped/cmd/kubeconfig.go

```diff
@@ -520,19 +537,130 @@ func discoverAuthenticatorParams(authenticator metav1.Object, flags *getKubeconf
 		}
 
 		// If the --oidc-ca-bundle flags was not set explicitly, default it to the
-		// spec.tls.certificateAuthorityData field of the JWTAuthenticator.
-		if len(flags.oidc.caBundle) == 0 && auth.Spec.TLS != nil && auth.Spec.TLS.CertificateAuthorityData != "" {
-			decoded, err := base64.StdEncoding.DecodeString(auth.Spec.TLS.CertificateAuthorityData)
+		// spec.tls.certificateAuthorityData field of the JWTAuthenticator, if that field is set, or else
+		// try to discover it from the spec.tls.certificateAuthorityDataSource, if that field is set.
+		if len(flags.oidc.caBundle) == 0 && auth.Spec.TLS != nil {
```

Comment:
I wonder if this `if` case should have an `else` just for logging to let the user know that we will choose not to include a CA bundle for the issuer in the resulting `kubeconfig`? (I know this isn't new for this PR, just a general comment).

Reply:
It's not necessarily a problem in the else case. For many OIDC providers, your system certs should be good enough and there should be no need to specify a CA bundle.